tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/CodeownersManualAnalysisTests.cs

```diff
@@ -23,124 +23,304 @@ namespace Azure.Sdk.Tools.RetrieveCodeOwners.Tests;
+    /// recipients will change once the new matcher is enabled.
+    /// The PR reviewers will remain unchanged, as they use the GitHub CODEOWNERS
+    /// interpreter which independent of ours and always supported wildcards.
+    /// </summary>
+    private void WriteMatcherDiffToCsv(
+        string targetDirPathSuffix,
+        string outputFilePrefix,
+        string ignoredPathPrefixes = Program.DefaultIgnoredPrefixes)
+    {
+        WriteOwnersDiffToCsv(
+            new[]
+            {
+                (targetDirPathSuffix, CodeownersFilePathSuffix, ignoredPathPrefixes, useRegexMatcher: false),
+                (targetDirPathSuffix, CodeownersFilePathSuffix, ignoredPathPrefixes, useRegexMatcher: true)
+            },
+            outputFilePrefix);
+    }
 
     #endregion
 
-    private static void WriteToFileOwnersData(
+    #region private static
+
+    /// <summary>
+    /// This method is similar to:
+    ///
+    ///     WriteOwnersDiffToCsv
+    ///
+    /// Except it is not doing any diffing: it just evaluates one invocation of
+    /// Azure.Sdk.Tools.RetrieveCodeOwners.Program.Main
+    /// and returns its information, in similar, but simplified table format.
+    ///
+    /// If given path, provided in column PATH, did not match any path in CODEOWNERS file,
+    /// the column PATH EXPRESSION will have a value of _____ .
+    ///
+    /// In addition, this method also does an validation of CODEOWNERS paths
+    /// and if it find a problem with given path, it returns output lines with ISSUE column
+    /// populated and PATH column empty, as there is no path to speak of - only CODEOWNERS path,
+    /// provided in PATH EXPRESSION column, is present.
+    ///
+    /// The ISSUE column has following codes:
+    ///
+    ///   INVALID_PATH_SHOULD_START_WITH_SLASH
+    ///     All CODEOWNERS paths must start with "/", but given path doesn't.
+    ///     Such path will still be processed by our CODEOWNERS interpreter, but nevertheless it is
+    ///     invalid and should be fixed.
+    /// 
+    ///   INVALID_PATH_SHOULD_HAVE_SUFFIX_SLASH_TO_DENOTE_DIR
+    ///     CODEOWNERS file contains a simple (i.e. without wildcards) path that is expected to match against
+    ///     a file, as it does not end with "/". However, the repository contains a directory with the same
+    ///     path. This means such CODEOWNERS path will never match any input path. Usually the proper fix
+    ///     is to add the missing suffix "/" to the path to make it correctly match against the existing directory.
+    /// 
+    ///   WILDCARD_FILE_PATH_NEEDS_MANUAL_EVAL
+    ///     Same situation as above, but the CODEOWNERS path is a file path with a wildcard, hence current
+    ///     validation implementation cannot yet determine if it should be a path to directory or not.
+    ///     Hence, this needs to be checked manually by ensuring that the wildcard file path matches
+    ///     at least one file in the repository.
+    /// </summary>
```

Comment:
Note to self: add here a catch-all rule for `INVALID_PATH`, where `CodeownersEntry.Valid` is false. Double-check it catches cases of incorrect usage of unsupported chars or `**`.